ingestion/src/metadata/sdk/data_quality/runner.py

```diff
@@ -0,0 +1,439 @@
+
+        config = OpenMetadataWorkflowConfig(**data)
+        source = config.source
+
+        assert (
+            source.type == TestSuiteConfigType.TestSuite.value
+        ), f"Can't create test suite for source type: {source.type}"
+
+        source_config = source.sourceConfig.config
+        assert isinstance(
+            source_config, TestSuitePipeline
+        ), f"Can't create test suite for source config type: {type(source.sourceConfig.config)}"
+        assert (
+            source_config.entityFullyQualifiedName is not None
+        ), "TestSuitePipeline config must have entity fully qualified name"
+
+        if use_connection_from_yaml:
+            client = OMeta(config=config.workflowConfig.openMetadataServerConfig)
+
+        runner = cls.for_table(
+            source_config.entityFullyQualifiedName.root, client=client
+        )
+
+        processor: Optional[TestSuiteProcessorConfig] = None
+        if config.processor and config.processor.config:
+            processor = TestSuiteProcessorConfig(**config.processor.config.model_dump())
+
+        if processor is None:
+            return runner
+
+        if tests_definitions := processor.testCases:
+            runner.test_definitions = tests_definitions
+
+        return runner
+
+    def _initialize(self) -> None:
+        """Fetch table entity and service connection from OpenMetadata.
+
+        This method retrieves the table entity and associated database service connection
+        from OpenMetadata. It validates that the table exists and has a properly configured
+        service connection.
+
+        Raises:
+            ValueError: If table not found, service not found, or connection not configured
+        """
+        self.table_entity = self.tables.get_by_name(
+            entity=Table,
+            fqn=self.table_fqn,
+            fields=[
+                "tableProfilerConfig",
+                "testSuite",
+                "serviceType",
+                "service",
+                "database",
+            ],
+        )
+
+        if self.table_entity is None:
+            raise ValueError(
+                f"Table '{self.table_fqn}' not found in OpenMetadata. "
+                + "Please verify the FQN is correct (format: service.database.schema.table)"  # noqa: E501
+            )
+        assert (
+            self.table_entity.service is not None
+        ), f"Service not found for table '{self.table_fqn}'"
+
+        service_id = self.table_entity.service.id
+        service = self.services.get_by_id(DatabaseService, service_id)
+
+        if service is None:
+            raise ValueError(f"Service '{service_id}' not found in OpenMetadata")
+
+        if service.connection is None:
+            raise ValueError(
+                f"Service '{service_id}' does not have a connection configured"
+            )
+
+        self.service_connection = service.connection
+
+    def add_test(self, test_definition: BaseTest) -> Self:
```

Comment:
If we think about that operation as something similar to python `append` let's not return the object.

e.g.
```
l = [1,2,3]
l.append(4) // this does not return the `l` object
```